flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/ExecNodeMetadataUtil.java

```diff
@@ -0,0 +1,294 @@
+    static void addTestNode(Class<? extends ExecNode<?>> execNodeClass) {
+        addToLookupMap(execNodeClass);
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static List<ExecNodeMetadata> extractMetadataFromAnnotation(
+            Class<? extends ExecNode> execNodeClass) {
+        List<ExecNodeMetadata> metadata = new ArrayList<>();
+        ExecNodeMetadata annotation = execNodeClass.getDeclaredAnnotation(ExecNodeMetadata.class);
+        if (annotation != null) {
+            metadata.add(annotation);
+        }
+
+        MultipleExecNodeMetadata annotations =
+                execNodeClass.getDeclaredAnnotation(MultipleExecNodeMetadata.class);
+        if (annotations != null) {
+            if (metadata.isEmpty()) {
+                for (ExecNodeMetadata annot : annotations.value()) {
+                    if (annot != null) {
+                        metadata.add(annot);
+                    }
+                }
+            } else {
+                throw new IllegalStateException(
+                        String.format(
+                                "ExecNode: %s is annotated both with %s and %s. Please use only "
+                                        + "%s or multiple %s",
+                                execNodeClass.getCanonicalName(),
+                                ExecNodeMetadata.class,
+                                MultipleExecNodeMetadata.class,
+                                MultipleExecNodeMetadata.class,
+                                ExecNodeMetadata.class));
+            }
+        }
+        return metadata;
+    }
+
+    private static void addToLookupMap(Class<? extends ExecNode<?>> execNodeClass) {
+        if (!JsonSerdeUtil.hasJsonCreatorAnnotation(execNodeClass)) {
+            throw new IllegalStateException(
+                    String.format(
+                            "ExecNode: %s does not implement @JsonCreator annotation on "
+                                    + "constructor.",
+                            execNodeClass.getCanonicalName()));
+        }
+
+        List<ExecNodeMetadata> metadata = extractMetadataFromAnnotation(execNodeClass);
+        if (metadata.isEmpty()) {
+            throw new IllegalStateException(
+                    String.format(
+                            "ExecNode: %s is missing %s annotation.",
+                            execNodeClass.getCanonicalName(),
+                            ExecNodeMetadata.class.getSimpleName()));
+        }
+
+        for (ExecNodeMetadata meta : metadata) {
+            doAddToMap(new ExecNodeNameVersion(meta.name(), meta.version()), execNodeClass);
+        }
+    }
+
+    private static void doAddToMap(
+            ExecNodeNameVersion key, Class<? extends ExecNode<?>> execNodeClass) {
+        if (LOOKUP_MAP.containsKey(key)) {
+            throw new IllegalStateException(
+                    String.format(
+                            "Found duplicate ExecNode: %s. This is a bug, please contact developers.",
```

Comment:
`This is a bug. Please file an issue.`

Reply:
I forgot to remove this, I think it's ok to completely remove the phrase `This is a bug....`